atom/browser/api/atom_api_window.cc

```diff
@@ -584,8 +584,17 @@ void Window::SetFocusable(bool focusable) {
   return window_->SetFocusable(focusable);
 }
 
-void Window::SetProgressBar(double progress) {
-  window_->SetProgressBar(progress);
+void Window::SetProgressBar(double progress, mate::Arguments* args) {
+  std::string mode;
+
+  if (args && args->Length() == 2) {
```

Comment:
What about using something like not_null<T> from GSL to indicated this?
https://github.com/Microsoft/GSL/blob/master/include/gsl.h#L75